workspace/__ardulib__/Graphics/XGraphics.h

```diff
@@ -0,0 +1,458 @@
+
+#ifndef X_GRAPHICS_H
+#define X_GRAPHICS_H
+
+#include "XRender.h"
+#include "XVector2.h"
+
+using xod::XColor;
+
+class XFill {
+private:
+public:
+    XFill() {}
+    virtual XColor colorAtPoint(XVector2<int16_t> point) {}
+    virtual void setSolidColor(XColor _color) {}
+};
+
+class XSolidFill : public XFill {
+private:
+    XColor _color;
+public:
+    XSolidFill(XColor color) {
+        _color = color;
+    }
+
+    XColor colorAtPoint(XVector2<int16_t> point);
+    void setSolidColor(XColor _color);
+};
+
+XColor XSolidFill::colorAtPoint(XVector2<int16_t> point) {
+    return _color;
+}
+
+void XSolidFill::setSolidColor(XColor color) {
+    _color = color;
+}
+
+class XBitmap {
+protected:
+    XBitmap* _parent;
+public:
+    XBitmap() {
+        _parent = nullptr;
+    }
+
+    virtual const XFill* getFill() const;
+    virtual const XColor* getBgColor() const;
```

Comment:
Does it make sense to use "fill" and "background-color" at the same time?
Maybe better to get rid of the background color and only fill and stroke will remain?

Reply:
Also, I would not expect any of those methods from a class named XBitmap. "XBitmap" sounds like a thing with known width and height that you can iterate over and get individual pixels, not something with fill, stroke etc